### DISCUSSION TITLE
Specify different layouts for routes within the same folder in the new layout system

### DISCUSSION THREAD

**Discussion by Nemmcy** [vercel/next.js#39576] [Ideas]
### Describe the feature you'd like to request

After reading the [Layout RFC blog post](https://nextjs.org/blog/layouts-rfc), I realized that the routes within the same folder will inevitably share the same layout.js file.

In most cases, a web app is hosted on a subdomain and has protected routes. Usually, the app homepage is on the root segment and almost all other pages are on the first non-root segment, protected or not. The problem is that those pages have different layouts which are wrapped by different route guards.

For instance, pages on '/' and '/analytics' are protected against non-authenticated users and share the same layout and pages on '/signup' and '/login' are protected against authenticated users and share the same layout. Ideally, we would like '/' and '/analytics' to share the root layout.js file, but at the same time we would like '/signup' and '/login' to share a totally different layout.js file which is currently not possible.

Creating a folder for each segment and adding a layout.js file will not work for two reasons:

1. The root layout.js file will still apply
2. There is no mechanism to tell Next.js that navigating between pages from different folders that have the same layout.js file should not re-render the layout

Currently, the only way to achieve this result is either by keep using getLayout on a per page basis or adding a single layout.js file at the root level of the app folder that displays a conditional layout using useRouter, which is not great.

### Describe the solution you'd like

1. Add the option to prevent layout.js from propagating further down the tree
2. Add the option to specify that the layout.js files from different folders are the same and make sure the layout will not re-render when navigating

### Describe alternatives you've considered

None.

**Comment by SukkaW** [comment]
As for your case, it could be implemented like this:

- Layout A (defined in `/user/layout.js`)
  - Route `/user/login`
  - Route `/user/signup`
- Layout B (defined in `/dashboard/layout.js`)
  - Route `/dashboard/`
  - Route `/dashboard/analytics`

And if you really want to keep `/signup`, `/login`, and `/analytics`, it is possible to use the Edge Middleware to dynamically rewrite the path (and do authentication).

**Comment by hamlim** [comment]
There will also be a way to handle this natively without needing to rewrite the path or use a nested url structure (if you go looking in the codebase you can find some examples), however I assume those will be covered in the second part of the RFC.

**Comment by leerob** [comment]
**Update:** We’re excited to announce [Next.js 13](https://nextjs.org/13), which includes the new `app/` directory (beta) with support for nested layouts, colocation of data fetching, streaming, Server Components by default, and much more.

Inside the `app/` directory, there is a powerful new way to fetch data with React’s `use()` hook and the extended Web `fetch()` API. This allows you to colocate data fetching directly inside components for the most flexibility, including inside layouts.

You can leave feedback here: https://github.com/vercel/next.js/discussions/41745
